subsys/caf/modules/buttons.c

```diff
@@ -156,7 +156,7 @@ static int callback_ctrl(bool enable)
 		/* Callbacks are disabled but they could fire in the meantime.
 		 * Make sure pending work is canceled.
 		 */
-		k_delayed_work_cancel(&button_pressed);
+		k_work_cancel_delayable(&button_pressed);
```

Comment:
This does not "make sure pending work is canceled".

Reply:
Can it fail if it's called from system workqueue context and the cancelled work also is scheduled for the system workqueue?

Reply:
Not if the work is scheduled or submitted but not running.  If the work is running it *will* fail, and this code has no idea what the status of the work item is.

Reply:
> If the work is running it will fail, and this code has no idea what the status of the work item is.

If cancel is called from system workqueue thread it cannot fail as work is never running (unless we cancel ourselves). It cannot be running as the code that is calling cancel is running (so a different work).
I think we are safe here.

Reply:
That's the sort of explanation that could be in a comment along with "Callbacks are disabled but they could fire in the meantime", to avoid repeating this sort of discussion.